plugin/pkg/auth/authorizer/node/node_authorizer.go

```diff
@@ -206,7 +206,7 @@ func (r *NodeAuthorizer) authorize(nodeName string, startingType vertexType, att
 
 	ok, err := r.hasPathFrom(nodeName, startingType, attrs.GetNamespace(), attrs.GetName())
 	if err != nil {
-		klog.V(2).Infof("NODE DENY: %v", err)
+		klog.V(2).InfoS("NODE DENY", "err", err)
```

Comment:
is "err" a standard name?

Reply:
do you have example output from this change?

Reply:
Yes, more details: https://github.com/kubernetes/community/blob/5e71635bb60dd7684e1afd820a1a61b9dd975211/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments

Reply:
> is "err" a standard name?

it seems to be. i used the "err" name because @serathius recommended that elsewhere: https://github.com/kubernetes/kubernetes/pull/91576#discussion_r432474929.

> do you have example output from this change?

ah, yeah, sorry, i forgot to put the output from this change in this pr.

before:
```
I0603 09:59:27.676862   54009 node_authorizer.go:209] NODE DENY: node "node2" cannot get secret ns1/node1-only, no relationship to this object was found in the node authorizer graph
```

after:
```
I0603 09:57:29.759232   53238 node_authorizer.go:209] "NODE DENY" err=&errors.errorString{s:"node \"node2\" cannot get secret ns1/node1-only, no relationship to this object was found in the node authorizer graph"}
```

Reply:
@ankeesler Can you use instructions provided herehttps://github.com/kubernetes/community/blob/5e71635bb60dd7684e1afd820a1a61b9dd975211/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#verify-log-output 
To generate comparison of log output?

Reply:
so for existing log consumers, this results in much less readable output:

> I0603 09:59:27.676862   54009 node_authorizer.go:209] NODE DENY: node "node2" cannot get secret ns1/node1-only, no relationship to this object was found in the node authorizer graph

vs

> I0603 09:57:29.759232   53238 node_authorizer.go:209] "NODE DENY" err=&errors.errorString{s:"node \"node2\" cannot get secret ns1/node1-only, no relationship to this object was found in the node authorizer graph"}

Reply:
This looks like a bug. We intended to use `%q` for errors.
Filed: https://github.com/kubernetes/klog/issues/158

Reply:
Is there action that needs to be taken on this PR in the interim before the bug is fixed?

I could pass `err.Error()` instead of `err`?

Reply:
We should wait until bug is fixed. We shouldn't pass `err.Error()` as it drops information about type that could be used by `InfoS` implementation.

Reply:
@ankeesler I have created PR with fix to serialization. https://github.com/kubernetes/kubernetes/pull/91792
Please to generate output with this change

Reply:
Using @serathius's branch from the above PR, here is what the output diff is.

`Infof`:
```
I0605 14:45:43.330939   41909 node_authorizer.go:209] NODE DENY: node "node2" cannot get secret ns1/node1-only, no relationship to this object was found in the node authorizer graph
```

`InfoS`:
```
I0605 14:43:40.047434   41115 node_authorizer.go:209] "NODE DENY" err="node \"node2\" cannot get secret ns1/node1-only, no relationship to this object was found in the node authorizer graph"
```

Reply:
the quote escaping is unfortunate... I didn't really expect non-json log output to change here, but if it's going to, we should avoid quoting things that will be escaped in ugly ways

Reply:
Quoting in InfoS here is needed to ensure logs can be parsed. I would be useful to move information stored in errors to logs (like node name). For this PR I think we should remove quotes used in error. 
@ankeesler can you change original error to not add quotes?

Reply:
> Quoting in InfoS here is needed to ensure logs can be parsed

when output as JSON, sure... when output as text, I didn't expect so

Reply:
After 0715ecfbb96, the structured log line looks like this.
```
I0605 15:14:45.707400   53254 node_authorizer.go:209] "NODE DENY" err="node node2 cannot get secret ns1/node1-only, no relationship to this object was found in the node authorizer graph"
```

Note - there was a pattern[0][1][2][3][4] of quoting the node name in log lines from this file. The log lines could potentially have different variations of quoting/not quoting the node name now. Is this OK?

[0] https://github.com/kubernetes/kubernetes/blob/b8b4186a14045ab66b150b5a92276d02b8a73a3e/plugin/pkg/auth/authorizer/node/node_authorizer.go#L210
[1] https://github.com/kubernetes/kubernetes/blob/b8b4186a14045ab66b150b5a92276d02b8a73a3e/plugin/pkg/auth/authorizer/node/node_authorizer.go#L213
[2] https://github.com/kubernetes/kubernetes/blob/b8b4186a14045ab66b150b5a92276d02b8a73a3e/plugin/pkg/auth/authorizer/node/node_authorizer.go#L238
[3] https://github.com/kubernetes/kubernetes/blob/b8b4186a14045ab66b150b5a92276d02b8a73a3e/plugin/pkg/auth/authorizer/node/node_authorizer.go#L308
[4] https://github.com/kubernetes/kubernetes/blob/b8b4186a14045ab66b150b5a92276d02b8a73a3e/plugin/pkg/auth/authorizer/node/node_authorizer.go#L313

Reply:
I guess replace those with `'%s'`?

Reply:
Gave that a go in 1b7352645bc.

Reply:
I would still wrap in single-quotes so it is clear which part is the name

Reply:
> here is where we are at with Infof logging.
>
> I0605 16:14:30.996413   76058 node_authorizer.go:281] NODE DENY: 'node0' authorizer.AttributesRecord{User:(*user.DefaultInfo)(0xc0003b16c0), Verb:"watch", Namespace:"", APIGroup:"storage.k8s.io", APIVersion:"", Resource:"csinodes", Subresource:"", Name:"", ResourceRequest:true, Path:""}

not sure I understand that... under what circumstances are we stringifying an interface pointer?

Reply:
edit: nevermind, those were two different lines. switching the quote to single quote looks fine to me